scripts/validate_docstrings.py

```diff
@@ -491,12 +492,14 @@ def first_line_ends_in_dot(self):
         if self.doc:
             return self.doc.split('\n')[0][-1] == '.'
 
+    @property
+    def deprecated_with_directive(self):
```

Comment:
Hi @datapythonista , what would be the best way to resolve this?

Reply:
Hmm understanding the reasoning more. Why do we have both properties though? Wouldn't just `deprecated` suffice?

Reply:
You're right, but as this is to validate that, we want to know if it's in the short summary, meaning that is the first thing in the docstring, to be able to report it as an error.

Reply:
So can't we just merge with `deprecated` anyway? As of now they are the same thing save the latter explicitly handling `Panel`.

Reply:
I don't see your point. We need both, `deprecated` for knowing if it's deprecated (when we generate the json, or when we use ``--ignore-deprecated`) and `deprecated_with_directive` to validate that the directive is in the correct position.

We could just leave `deprecated_with_directive` and check in two places also whether it's a method of `Panel` (and in the future may be `SparseDataFrame`...). But that seems a much worse approach, duplicating code and keeping things clear.

Am I not understanding you?

Reply:
I think we are conceptually there but there are parts of the implementation that I think are going to make this confusing. Specifically assuming that the special casing for Panel is temporary, then `deprecated` and `deprecated_with_directive` are the same thing. So if the point is to do it in two passes to:

1. First figure out what docstrings are deprecated AND then
2. Ensure that the directive is placed appropriately therein

The current implementation doesn't really do that, because again ignoring the special casing for Panel `directive` and `deprecated_with_directive` are the same exact thing.

Are you expecting the implementation of `directive` to change in a separate PR?

Reply:
I don't understand well, but I don't want to spend more time in a minor thing like this. If you can tell @aqurilla what you propose, so it can be implemented, that would be great.

Reply:
Yea @aqurilla if you can combine the two methods into just `deprecated` and use that; I don't think there's a need for two separate properties

Reply:
That's what was first implemented, and the validation will return every method of `Panel` as a false positive for not having the `.. deprecate::` directive in the right position. That's what I tried to explain before.

Reply:
Should I revert the implementation to the earlier one in this case?